
From: Andi Kleen <ak@muc.de>

Add an kstack= option to configure how much stack should be printed on a
oops.

Signed-off-by: Andi Kleen <ak@suse.de>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/arch/x86_64/kernel/traps.c |    8 ++++++++
 1 files changed, 8 insertions(+)

diff -puN arch/x86_64/kernel/traps.c~add-an-option-to-configure-oops-stack-dump-on-x86-64 arch/x86_64/kernel/traps.c
--- 25/arch/x86_64/kernel/traps.c~add-an-option-to-configure-oops-stack-dump-on-x86-64	Tue Oct  5 14:37:03 2004
+++ 25-akpm/arch/x86_64/kernel/traps.c	Tue Oct  5 14:37:03 2004
@@ -903,3 +903,11 @@ static int __init oops_dummy(char *s)
 	return -1; 
 } 
 __setup("oops=", oops_dummy); 
+
+static int __init kstack_setup(char *s)
+{
+	kstack_depth_to_print = simple_strtoul(s,NULL,0);
+	return 0;
+}
+__setup("kstack=", kstack_setup);
+
_
